stest/assertions.go

```diff
@@ -153,14 +166,16 @@ func ObserveSupervisor(
 	// We always want to start the supervisor for test purposes, so this is
 	// embedded in the ObserveSupervisor call
 	sup, err := spec.Start(ctx)
-	if err != nil {
-		return []s.Event{}, err
-	}
 
 	evIt := evManager.Iterator()
 
 	// Make sure all the tree started before doing assertions
-	evIt.SkipTill(ProcessStarted(rootName))
+	evIt.SkipTill(ProcessName(rootName))
```

Comment:
Add comment around why we do the `SkipTill` before returning the error

Reply:
Also, add comment around why we use `ProcessName` instead of `ProcessStarted` here